minecraft/networking/connection.py

```diff
@@ -436,14 +436,23 @@ def read_packet(self, stream):
 
             if self.connection.options.compression_enabled:
                 decompressed_size = VarInt.read(packet_data)
-                if decompressed_size > 0:
-                    decompressed_packet = decompress(packet_data.read())
-                    assert len(decompressed_packet) == decompressed_size, \
-                        'decompressed length %d, but expected %d' % \
-                        (len(decompressed_packet), decompressed_size)
-                    packet_data.reset()
-                    packet_data.send(decompressed_packet)
-                    packet_data.reset_cursor()
+                rawPacketData = None
+                try:
```

Comment:
I'm not convinced this is the right way to handle this, can you provide more information about when this problem happens? What server with what plugin configuration?